src/main/scala/main.scala

```diff
@@ -54,4 +52,27 @@ object RockSlicer {
     jsonBlocks.saveAsTextFile("blocks.json")
     sc.stop()
   }
+
+  // Produce joints that achieve some lead balancing when generating the initial Block RDD
+  private def generateSeedJoints(joints: Seq[Joint], numSeedJoints: Integer): (Seq[Joint], Seq[Joint]) = {
+    val (persistentJoints, nonPersistentJoints) = joints.partition(_.shape.isEmpty)
+    persistentJoints.length match {
+      // All of the seed joints will be persistent
+      case l if l >= numSeedJoints => (persistentJoints.take(numSeedJoints),
+          persistentJoints.takeRight(l - numSeedJoints) ++ nonPersistentJoints)
+      // All persistent joints are used as seed joints, along with some non-persistent joints
+      case l =>
+          // Sort non-persistent joints by relative position along x axis
+          val sortedNonPersistent = nonPersistentJoints.sortWith(_.localOrigin._1 < _.localOrigin._1)
```

Comment:
Use `center` instead of `localOrigin` - all joints will have the same `locaOrigin` initially.